test/CMakeTests.cmake

```diff
@@ -833,7 +833,7 @@ if (BUILD_SHARED_LIBS)
 
   add_test (NAME H5PLUGIN-filter_plugin COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:filter_plugin>)
   set_tests_properties (H5PLUGIN-filter_plugin PROPERTIES
-      ENVIRONMENT "HDF5_PLUGIN_PATH=${CMAKE_BINARY_DIR}/filter_plugin_dir1${CMAKE_SEP}${CMAKE_BINARY_DIR}/filter_plugin_dir2;srcdir=${HDF5_TEST_BINARY_DIR}"
+      ENVIRONMENT "HDF5_PLUGIN_PATH=${CMAKE_BINARY_DIR}/filter_plugin_dir1${CMAKE_SEP}${CMAKE_BINARY_DIR}/filter_plugin_dir2;srcdir=${HDF5_TEST_BINARY_DIR};HDF5_VOL_CONNECTOR="
```

Comment:
Hmm I think this should rather go the other way around. If a user sets `HDF5_VOL_CONNECTOR`, we should assume they know what they are doing and may have set `HDF5_PLUGIN_PATH` as well, in which case the tests that modify `HDF5_PLUGIN_PATH` should be appending on to anything that exists rather than overriding it.